Applications/ApplicationsLib/ProjectData.h

```diff
@@ -134,6 +136,9 @@ class ProjectData final
     /// Checks if a parameter has name tag.
     void parseParameters(BaseLib::ConfigTree const& parameters_config);
 
+    /// Parses material configuration and saves them in an object.
+    void parseMaterials(boost::optional<BaseLib::ConfigTree> const& media_config);
```

Comment:
Why do you use `boost::optional`? As far as I understand material configuration is required. Or is it just a temporary solution until all processes are changed?

Reply:
Indeed. As long as the method does not cover all existing material properties, the declaration of the material object is obtional-